cg/meta/workflow/nf_analysis.py

```diff
@@ -307,3 +320,109 @@ def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
                 )
             files.append(FileDeliverable(**file))
         return PipelineDeliverables(files=files)
+
+    def get_multiqc_json_path(self, case_id: str) -> Path:
+        """Return the path of the multiqc_data.json file."""
+        return Path(
+            self.root_dir,
+            case_id,
+            MultiQC.MULTIQC,
+            MultiQC.MULTIQC_DATA,
+            MultiQC.MULTIQC_DATA + FileExtensions.JSON,
+        )
+
+    def get_pipeline_metrics(self) -> dict:
+        """Get nf-core pipeline metrics constants."""
+        return {}
+
+    def get_multiqc_json_metrics(self, case_id: str) -> list[MetricsBase]:
+        """Get a multiqc_data.json file and returns metrics and values formatted."""
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        sample_id: str = case.links[0].sample.internal_id
+        multiqc_json = MultiqcDataJson(
+            **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
+        )
+        metrics_values: dict = {}
+
+        for key in multiqc_json.report_general_stats_data:
+            if case_id in key:
+                metrics_values.update(list(key.values())[0])
+                LOG.info(f"Key: {key}, Values: {list(key.values())[0]}")
```

Comment:
you can make the first part of this function into a smaller function:
```python
def get_multiqc_metric_values_from_case_id(case_id: str) -> dict:
    """<add docstring here>"""
    multiqc_json: dict = MultiqcDataJson(
        **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
    ).report_general_stats_data
    metrics_values: dict = {}
    for key in multiqc_json:
        if case_id in key:
            metrics_values.update(list(key.values())[0])
            LOG.info(f"Key: {key}, Values: {list(key.values())[0]}")
    return metrics_values
```
and the second part:
```python
def get_metric_base_list(sample_id: str, metrics_values: dict) -> list[MetricsBase]:
    """<add_docstring_here>"""
    metric_base_list: list[MetricsBase] = []
    for metric_name, metric_value in metrics_values.items():
        metric_base_list.append(
            MetricsBase(
                header=None,
                id=sample_id,
                input=MultiQC.MULTIQC_DATA + FileExtensions.JSON,
                name=metric_name,
                step=MultiQC.MULTIQC,
                value=metric_value,
                condition=self.get_pipeline_metrics().get(metric_name, None),
            )
        )    
    return metric_base_list
```
then the final function is:
```python
def get_multiqc_json_metrics(self, case_id: str) -> list[MetricsBase]:
    """<add_docstring>"""
    case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
    sample_id: str = case.links[0].sample.internal_id
    metric_values: dict = self.get_multiqc_metric_values_from_case_id(case_id)
    metric_base_list: list = get_metric_base_list(sample_id=sample_id, metrics_values=metric_values)
    return metric_base_list
```